src/main/telemetry/ibus.c

```diff
@@ -0,0 +1,387 @@
+static serialPortConfig_t *ibusSerialPortConfig;
+
+static bool ibusTelemetryEnabled = false;
+static portSharing_e ibusPortSharing;
+
+#define INVALID_IBUS_ADDRESS 0
+static ibusAddress_t ibusBaseAddress = INVALID_IBUS_ADDRESS;
+
+static uint8_t ibusReceiveBuffer[IBUS_RX_BUF_LEN] = { 0x0 };
+
+static uint16_t calculateChecksum(uint8_t ibusPacket[static IBUS_CHECKSUM_SIZE], size_t packetLength) {
+    uint16_t checksum = 0xFFFF;
+    for (size_t i = 0; i < packetLength - IBUS_CHECKSUM_SIZE; i++) {
+        checksum -= ibusPacket[i];
+    }
+
+    return checksum;
+}
+
+static bool isChecksumOk(uint8_t ibusPacket[static IBUS_CHECKSUM_SIZE], size_t packetLength, uint16_t calcuatedChecksum) {
+    // Note that there's a byte order swap to little endian here
+    return (calcuatedChecksum >> 8) == ibusPacket[packetLength - 1]
+            && (calcuatedChecksum & 0xFF) == ibusPacket[packetLength - 2];
+}
+
+static void transmitIbusPacket(uint8_t ibusPacket[static IBUS_MIN_LEN], size_t packetLength) {
+    uint16_t checksum = calculateChecksum(ibusPacket, packetLength);
+    ibusPacket[packetLength - IBUS_CHECKSUM_SIZE] = (checksum & 0xFF);
+    ibusPacket[packetLength - IBUS_CHECKSUM_SIZE + 1] = (checksum >> 8);
+    for (size_t i = 0; i < packetLength; i++) {
+        serialWrite(ibusSerialPort, ibusPacket[i]);
+    }
+}
+
+static void sendIbusCommand(ibusAddress_t address) {
+    uint8_t sendBuffer[] = { 0x04, IBUS_COMMAND_DISCOVER_SENSOR | address, 0x00, 0x00 };
+    transmitIbusPacket(sendBuffer, sizeof sendBuffer);
+}
+
+static void sendIbusSensorType(ibusAddress_t address) {
+    uint8_t sendBuffer[] = {0x06,
+                            IBUS_COMMAND_SENSOR_TYPE | address,
+                            SENSOR_ADDRESS_TYPE_LOOKUP[address - ibusBaseAddress],
+                            0x02, 0x0, 0x0 };
+    transmitIbusPacket(sendBuffer, sizeof sendBuffer);
+}
+
+static void sendIbusMeasurement(ibusAddress_t address, uint16_t measurement) {
+    uint8_t sendBuffer[] = { 0x06, IBUS_COMMAND_MEASUREMENT | address, measurement & 0xFF, measurement >> 8, 0x0, 0x0 };
+    transmitIbusPacket(sendBuffer, sizeof sendBuffer);
+}
+
+static bool isCommand(ibusCommand_e expected, uint8_t ibusPacket[static IBUS_MIN_LEN]) {
+    return (ibusPacket[1] & 0xF0) == expected;
+}
+
+static ibusAddress_t getAddress(uint8_t ibusPacket[static IBUS_MIN_LEN]) {
+    return (ibusPacket[1] & 0x0F);
+}
+
+static void dispatchMeasurementRequest(ibusAddress_t address) {
+    switch (SENSOR_ADDRESS_TYPE_LOOKUP[address - ibusBaseAddress]) {
+    case IBUS_SENSOR_TYPE_EXTERNAL_VOLTAGE:
+        {
+            uint16_t value = vbat * 10;
+            if (ibusTelemetryConfig()->report_cell_voltage) {
+                value /= batteryCellCount;
+            }
+            sendIbusMeasurement(address, value);
+        }
+        break;
+
+    case IBUS_SENSOR_TYPE_TEMPERATURE:
+        {
+            #ifdef BARO
+                float temperature = (baroTemperature + 50) / 100.f;
```

Comment:
This looks like integer rounding - but it will not work for floats (and will work incorrectly for negative values)

If you want to round the value, do so explicitly (`roundf()`) after final scaling.